src/validation/validator.ts

```diff
@@ -0,0 +1,51 @@
+import { ValidationFactory } from './factory'
+import { FieldValidator } from './fieldValidators'
+import { Severity, Validation, ValidationResult } from './validation'
+
+export type FieldsValidators = {
+  [prop: string]: Array<FieldValidator>
+}
+
+interface ValidateOptions {
+  removeValidFields: boolean
+}
+
+const defaultValidateOptions: ValidateOptions = { removeValidFields: true }
+
+export class Validator {
+  private async validateProp(obj: any, prop: string, fieldValidators: Array<FieldValidator>): Promise<Validation> {
+    const validations = await Promise.all(fieldValidators.map((fieldValidator) => fieldValidator(prop, obj)))
+    const errors: Array<ValidationResult> = []
+    const warnings: Array<ValidationResult> = []
+    validations.forEach((validationResult) => {
+      if (validationResult) {
+        const arr = validationResult.severity === Severity.error ? errors : warnings
+        arr.push(validationResult)
+      }
+    })
+    const valid = errors.length === 0 && warnings.length === 0
+    return ValidationFactory.createInstance({ valid, errors, warnings })
+  }
+
+  async validate(
+    obj: any,
+    propsValidators: FieldsValidators,
```

Comment:
call this parameter fieldValidators:{[name:string]: FieldValidator}

Reply:
ok